Modules/Module_Import/xASL_wrp_DCM2NII_Subject.m

```diff
@@ -103,6 +103,12 @@
             end
             
             %% 4. Iterate over scans
+			% Safety-check, we can't allow multiple matches for the same sequence
+			if numel(thisRun.scanIDs) ~= numel(unique(thisRun.scanIDs))
+				fprintf('Detected duplicit scans: %s\n',strjoin(thisRun.scanIDs, ', '));
+				error('We do not allow multiple matches per scan for. Please delete some sequences or adjust sourcestructure.json');
```

Comment:
should be

error('We do not allow multiple token matches per scantype. Please adjust sourcestructure.json');

For clarity, but most importantly users should NOT delete sourcedata! Ideally this is always readonly, because from the philosophy of BIDS sourcedata is always exactly what you got, without restructuring etc. I know this is not what we do yet for practical reasons, but you get my point :)

Reply:
good point. fixed.